tools/validate-deployment.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# This script verifies the successfull deployment of the resources needed for the DevOps OpenHack.
+# You need to provide the CSV file with all the credentials of the Azure subscriptions from the classroom management portal and a private / public SSH keypair that will be used to access the provisioning VMs
+# The error log file is where will be logged the informations regarding the failed deployments. If not provided, it defaults to error.log. 
+
+usage() { echo "Usage: getteams.sh -f <errorlog_file> -u <service principal username> -f < service principal password> -x < service principal tenant>" 1>&2; exit 1; }
+
+while getopts ":f:k:p:u:x:t:" arg; do
+    case "${arg}" in
+        f)
+            ERROR_FILE=${OPTARG}
+        ;;
+        u)
+            USERNAME=${OPTARG}
+        ;;
+        x)
+            PASSWORD=${OPTARG}
+        ;;
+        t)
+            TENANT=${OPTARG}
+        ;;
+    esac
+done
+
+KEY_DIR=~/devopsohkeys
+mkdir -p ${KEY_DIR}
+ID_RSA_PRIVATE=${KEY_DIR}/devops_openhack_key
+ID_RSA_PUBLIC=${KEY_DIR}/devops_openhack_key.pub
+ssh-keygen -t rsa -C "DevOps OpenHack" -f ${ID_RSA_PRIVATE} -P ""
+if [ $? -ne 0 ]; then
+    echo "[ERROR] Creating the ssh keypair for the proctorVM failed "
+fi
+
```

Comment:
The code that aims at reusing a set of private/public key pair can be removed since this is going to be run on a new deployment each time.